tools/src/main/scala/scala/scalanative/build/Discover.scala

```diff
@@ -174,6 +178,39 @@ object Discover {
     path
   }
 
+  /** Detect the target architecture.
+   *
+   *  @param clang
+   *    A path to the executable `clang`.
+   *  @param workdir
+   *    A working directory where the compilation will take place.
+   *  @return
+   *    The detected target triple describing the target architecture.
+   */
+  def targetTriple(clang: Path, workdir: Path): String = {
+    // Use non-standard extension to not include the ll file when linking (#639)
+    val targetc = workdir.resolve("target").resolve("c.probe")
```

Comment:
We removed the probe code in another PR. The target triple is detected via the normal tool chain or via the NativeConfig setting. This may be a merge issue.